tests/gradients/test_parameter_shift_cv.py

```diff
@@ -0,0 +1,1105 @@
+
+    @pytest.mark.parametrize("t", [0, 1])
+    def test_interferometer(self, t, tol):
+        """An integration test for CV gates that support analytic differentiation
+        if succeeding the gate to be differentiated, but cannot be differentiated
+        themselves (for example, they may be Gaussian but accept no parameters,
+        or may accept a numerical array parameter.).
+
+        This ensures that, assuming their _heisenberg_rep is defined, the quantum
+        gradient analytic method can still be used, and returns the correct result.
+
+        Currently, the only such operation is qml.Interferometer. In the future,
+        we may consider adding a qml.GaussianTransfom operator.
+        """
+
+        if t == 1:
+            pytest.xfail(
+                "There is a bug in the second order CV parameter-shift rule; "
+                "phase arguments return the incorrect derivative."
+            )
+
+            # Note: this bug currently affects PL core as well:
+            #
+            # dev = qml.device("default.gaussian", wires=2)
+            #
+            # U = np.array([[ 0.51310276+0.81702166j,  0.13649626+0.22487759j],
+            #         [ 0.26300233+0.00556194j, -0.96414101-0.03508489j]])
+            #
+            # @qml.qnode(dev)
+            # def circuit(r, phi):
+            #     qml.Displacement(r, phi, wires=0)
+            #     qml.Interferometer(U, wires=[0, 1])
+            #     return qml.expval(qml.X(0))
+            #
+            # r = 0.543
+            # phi = 0.
+            #
+            # >>> print(circuit.jacobian([r, phi], options={"force_order2":False}))
+            # [[ 1.02620552 0.14823494]]
+            # >>> print(circuit.jacobian([r, phi], options={"force_order2":True}))
+            # [[ 1.02620552 -0.88728552]]
+
+        U = np.array(
+            [
+                [0.51310276 + 0.81702166j, 0.13649626 + 0.22487759j],
+                [0.26300233 + 0.00556194j, -0.96414101 - 0.03508489j],
+            ]
+        )
+
+        with qml.tape.JacobianTape() as tape:
+            qml.Displacement(0.543, 0, wires=0)
+            qml.Interferometer(U, wires=[0, 1])
+            qml.expval(qml.X(0))
+
+        tape.trainable_params = {t}
+
+        dev = qml.device("default.gaussian", wires=2)
+
+        tapes, fn = qml.gradients.finite_diff(tape)
+        grad_F = fn(dev.batch_execute(tapes))
+
+        tapes, fn = param_shift_cv(tape, dev)
+        grad_A = fn(dev.batch_execute(tapes))
+
+        tapes, fn = param_shift_cv(tape, dev, force_order2=True)
+        grad_A2 = fn(dev.batch_execute(tapes))
+
+        assert tape._par_info[0]["grad_method"] == "A"
+        assert tape._par_info[1]["grad_method"] == "A"
+
+        # the different methods agree
+        assert np.allclose(grad_A, grad_F, atol=tol, rtol=0)
+        assert np.allclose(grad_A2, grad_F, atol=tol, rtol=0)
+
+
+class TestVarianceQuantumGradients:

```

Comment:
Copy and paste of https://github.com/PennyLaneAI/pennylane/blob/ab71001bc55c2945a0945f42131c45d77da3512e/tests/tape/test_cv_param_shift.py#L702